CONTRIBUTING.md

```diff
@@ -0,0 +1,291 @@
+* Being responsive to feedback on pull requests. This means incorporating or
+  responding to all suggested changes, and leaving a note if you won't be
+  able to address things within a few days.
+* Being helpful and friendly on chat.zulip.org.
+
+These are also the main criteria we use to select interns for all of our
+internship programs.
+
+## Reporting issues
+
+If you find an easily reproducible bug and/or are experienced in reporting
+bugs, feel free to just open an issue on the relevant project on GitHub.
+
+If you have a feature request or are not yet sure what the underlying bug
+is, the best place to post issues is
+[#issues](https://chat.zulip.org/#narrow/stream/issues) (or
+[#mobile](https://chat.zulip.org/#narrow/stream/mobile) or
+[#electron](https://chat.zulip.org/#narrow/stream/electron)) on the
+[Zulip community server](http://zulip.readthedocs.io/en/latest/chat-zulip-org.html).
+This allows us to interactively figure out what is going on, let you know if
+a similar issue has already been opened, and collect any other information
+we need. Choose a 2-4 word topic that describes the issue, explain the issue
+and how to reproduce it if known, your browser/OS if relevant, and a
+[screenshot or screenGIF](http://zulip.readthedocs.io/en/latest/screenshot-and-gif-software.html)
+if appropriate.
+
+**Reporting security issues**. Please do not report security issues
+  publicly, including on public streams on chat.zulip.org. You can email
+  zulip-security@googlegroups.com. We create a CVE for every security issue.
+
+## User feedback
+
+Nearly every feature we develop starts with a user request. If you are part
+of a group that is either using or considering using Zulip, we would love to
+hear about your experience with the product. If you're not sure what to
+write, here are some questions we're always very curious to know the answer
+to:
+
+* Evaluation: What is the process by which your organization chose or will
+  choose a group chat product?
+* Pros and cons: What are the pros and cons of Zulip for your organization,
+  and the pros and cons of other products you are evaluating?
+* Features: What are the features that are most important for your
+  organization? In the best case scenario, what would your chat solution do
+  for you?
+* Onboarding: If you remember it, what was your impression during your first
+  few minutes of using Zulip? What did you notice, and how did you feel? Was
+  there anything that stood out to you as confusing, or broken, or great?
+* Organization: What does your organization do? How big is the organization?
+  A link to your organization's website?
+
+## Internship programs
+
+Zulip runs internship programs with [Outreachy](https://www.outreachy.org/),
+[Google Summer of Code](https://developers.google.com/open-source/gsoc/),
+and the
+[MIT Externship program](https://alum.mit.edu/students/NetworkwithAlumni/ExternshipProgram),
+and also takes summer interns from Harvard, MIT, and Stanford.
```

Comment:
s/also takes/has in the past taken/.  I don't want to make it sound like a requirement.